src/main/java/de/tum/cit/aet/artemis/programming/service/localvc/LocalVCServletService.java

```diff
@@ -704,6 +709,45 @@ private Commit extractCommitInfo(String commitHash, Repository repository) throw
         return new Commit(commitHash, author.getName(), revCommit.getFullMessage(), author.getEmailAddress(), branch);
     }
 
+    /**
+     * Retrieves the user from the HTTP request's authorization header.
+     *
+     * @param request the {@link HttpServletRequest} containing the authorization header.
+     * @return the {@link User} associated with the login found in the authorization header.
+     * @throws LocalVCAuthException if the user cannot be found or authorization fails.
+     */
+    private User getUserFromRequest(HttpServletRequest request) throws LocalVCAuthException {
+        String authorizationHeader = request.getHeader(LocalVCServletService.AUTHORIZATION_HEADER);
+
+        UsernameAndPassword usernameAndPassword = extractUsernameAndPassword(authorizationHeader);
+        return userRepository.findOneByLogin(usernameAndPassword.username()).orElseThrow(LocalVCAuthException::new);
+    }
+
+    /**
+     * Retrieves the participation for a programming exercise based on the repository URI.
+     *
+     * @param localVCRepositoryUri the {@link LocalVCRepositoryUri} containing details about the repository.
+     * @return the {@link ProgrammingExerciseParticipation} corresponding to the repository URI.
+     */
+    private ProgrammingExerciseParticipation getExerciseParticipationFromLocalVCRepositoryUri(LocalVCRepositoryUri localVCRepositoryUri) {
+        String projectKey = localVCRepositoryUri.getProjectKey();
+        String repositoryTypeOrUserName = localVCRepositoryUri.getRepositoryTypeOrUserName();
+        ProgrammingExercise exercise = getProgrammingExerciseOrThrow(projectKey);
```

Comment:
How often is this method called? Should we try to avoid this database call?

Reply:
Basically its called once per git operation. Maybe there's a way to fetch the participation directly from the URL, without needing to query for the exercise?

Reply:
I added a simpler database call, which fetches the participation directly from the repositoryURL. That obsoletes fetching the exercise, and then the participation